src/util.cpp

```diff
@@ -105,9 +105,84 @@ std::string to_internal(const std::string &);
 
 using namespace std;
 
+namespace Logging {
+//Globals defined here because link fails if in globals.cpp.
+//Keep at top of file so init first:
+	uint64_t categoriesEnabled = 0; //64 bit log id mask.
+	static map<uint64_t, std::string> logLabelMap = LOGLABELMAP; //Lookup log label from log id.
+
+
+uint64_t LogFindCategory(const std::string label)
+{
+	for (auto& x: logLabelMap)
+	{
+		if((std::string)x.second == label)
+			return (uint64_t)x.first;
+	}
+	return NONE;
+}
+
+std::string LogGetLabel(uint64_t category)
+{
+	string label = "none";
+	if(logLabelMap.count(category)!=0)
+		label = logLabelMap[category];
+
+	return label;
+}
+
+int LogWriteStr(const std::string &str)
+{
+	return LogPrintStr(str);
+}
```

Comment:
Can LogWriteStr be removed and call LogPrintStr directly?

Reply:
seconded

Reply:
All log related functions should exist in the Logging name space. 
LogPrintStr() does not.  (It is old.)
I think LogPrintStr() should be moved to LogWriteStr() in the Logging name space.
All calls LogPrintStr() should be changed to Logging::LogWriteStr()

LMK if you agree/disagree.
If not then I will do what you ask.

Reply:
please remove LogWriteStr.  Next cleanup, we'll work on moving LogPrintStr into the namespace.